File: languages/java/app-encryption/src/test/java/com/godaddy/asherah/appencryption/testhelpers/dummy/DummyCryptoPolicy.java
package com.godaddy.asherah.appencryption.testhelpers.dummy;

import java.time.Instant;

import com.godaddy.asherah.crypto.CryptoPolicy;

public class DummyCryptoPolicy implements CryptoPolicy {

  @Override
  public boolean isKeyExpired(Instant keyCreationDate) {
    return false;
  }

  @Override
  public long getRevokeCheckPeriodMillis() {
    return Long.MAX_VALUE;
  }

  @Override
  public boolean canCacheSystemKeys() {
    return true;
  }

  @Override
  public boolean canCacheIntermediateKeys() {
    return true;
  }

  @Override
  public boolean notifyExpiredIntermediateKeyOnRead() {
    return true;
  }

  @Override
  public boolean notifyExpiredSystemKeyOnRead() {
    return true;
  }

  @Override
  public KeyRotationStrategy keyRotationStrategy() {
    return KeyRotationStrategy.INLINE;
  }

  @Override
  public boolean isInlineKeyRotation() {
    return true;
  }

  @Override
  public boolean isQueuedKeyRotation() {
    return false;
  }

  @Override
  public String toString() {
    return getClass().getSimpleName() + "[policy=NeverExpire]";
  }
}

File: csharp/AppEncryption/AppEncryption.Tests/AppEncryption/TestHelpers/Dummy/DummyCryptoPolicy.cs
using System;
using GoDaddy.Asherah.Crypto;

namespace GoDaddy.Asherah.AppEncryption.Tests.AppEncryption.TestHelpers.Dummy
{
    public class DummyCryptoPolicy : CryptoPolicy
    {
        public override bool IsKeyExpired(DateTimeOffset keyCreationTime)
        {
            return false;
        }

        public override long GetRevokeCheckPeriodMillis()
        {
            return long.MaxValue;
        }

        public override bool CanCacheSystemKeys()
        {
            return true;
        }

        public override bool CanCacheIntermediateKeys()
        {
            return true;
        }

        public override bool CanCacheSessions()
        {
            return false;
        }

        public override long GetSessionCacheMaxSize()
        {
            return long.MaxValue;
        }

        public override long GetSessionCacheExpireMillis()
        {
            return long.MaxValue;
        }

        public override bool NotifyExpiredIntermediateKeyOnRead()
        {
            return true;
        }

        public override bool NotifyExpiredSystemKeyOnRead()
        {
            return true;
        }

        public override KeyRotationStrategy GetKeyRotationStrategy()
        {
            return KeyRotationStrategy.Inline;
        }

        public override bool IsInlineKeyRotation()
        {
            return true;
        }

        public override bool IsQueuedKeyRotation()
        {
            return false;
        }

        public override string ToString()
        {
            return typeof(DummyCryptoPolicy).FullName + "[policy=NeverExpire]";
        }
    }
}
